javascript-client/test/api/PartApi.spec.js

```diff
@@ -0,0 +1,141 @@
+ * https://github.com/swagger-api/swagger-codegen.git
+ *
+ * Swagger Codegen version: 2.4.6
+ *
+ * Do not edit the class manually.
+ *
+ */
+let expect = require("chai").expect;
+(function (root, factory) {
+  if (typeof define === 'function' && define.amd) {
+    // AMD.
+    define(['expect.js', '../../src/index'], factory);
+  } else if (typeof module === 'object' && module.exports) {
+    // CommonJS-like environments that support module.exports, like Node.
+    factory(require('expect.js'), require('../../src/index'));
+  } else {
+    // Browser globals (root is window)
+    factory(root.expect, root.SwaggerTunningPlace);
+  }
+}(this, function (expect, SwaggerTunningPlace) {
+  'use strict';
+
+  var instance;
+
+  beforeEach(function () {
+    instance = new SwaggerTunningPlace.PartApi();
+    instance.apiClient.basePath = 'http://localhost:5000';
+  });
+
+  var getProperty = function (object, getter, property) {
+    // Use getter method if present; otherwise, get the property directly.
+    if (typeof object[getter] === 'function')
+      return object[getter]();
+    else
+      return object[property];
+  }
+
+  var setProperty = function (object, setter, property, value) {
+    // Use setter method if present; otherwise, set the property directly.
+    if (typeof object[setter] === 'function')
+      object[setter](value);
+    else
+      object[property] = value;
+  }
+
+  describe('PartApi', function () {
+    // describe('addPart', function () {
+    //   it('should call addPart successfully', function (done) {
+    //     let part = {
+    //       "title": "test title1",
+    //       "model": "test model1",
+    //       "year": 2018,
+    //       "description": "This is test description. All-in-one kit",
+    //       "price": 50,
+    //       "image": "image1",
+    //     }
+    //     instance.addPart(part, function (error, data, res) {
+    //       if (error) throw error;
+    //       expect(res.status).to.be(200);
+    //       done();
+    //     });
+    //   });
+    //   it('should return bad request', function (done) {
+    //     let part = {
+    //       "model": "test model1",
+    //       "year": 2018,
+    //       "description": "This is test description. All-in-one kit",
+    //       "price": 50,
+    //       "image": "image1",
+    //     }
+    //     instance.addPart(part, function (error, data, res) {
+    //       expect(error.message).to.be("Bad Request");
+    //       done();
+    //     });
+    //   });
+    //   it('should return form errors', function (done) {
+    //     let part = {
+    //       "title": "",
+    //       "model": "test model1",
+    //       "year": 2018,
+    //       "description": "This is test description. All-in-one kit",
+    //       "price": 50,
+    //       "image": "image1",
+    //     }
+    //     instance.addPart(part, function (error, data, res) {
+    //       expect(res.body.message).to.be("Check the form for errors.");
```

Comment:
In general you should validate everywhere the status code. This way you will ensure that API returns proper status code to the clients.
Checking the error message might be tricky especially if you have multilanguage error message. Also error messages are subject to change. That is way it is much better a errorCode to be returned along with the error message. ErrorCode is better because it doesn't change and you can validate it easliy.